src/internal/m365/exchange/contacts_restore.go

```diff
@@ -100,6 +99,14 @@ func restoreContact(
 	errs *fault.Bus,
 	cb *count.Bus,
 ) (*details.ExchangeInfo, error) {
+	// contacts has a weird relationship with its default
+	// folder, which is that the folder is treated as invisible
+	// in many cases.  If we're restoring to a blank location,
+	// we can interpret that as the root.
```

Comment:
It's graph, yeah, the problem lies in a difference between the folder lookup and item destination.  It's kinda like how with emails, the folders aren't all under `Inbox`, they're often under `msgfolderroot`.  The difference there is that the messages themselves are only ever inside a folder that's a child of msgfolderroot.  Contacts can attach contact items directly to its root, instead of requiring a child folder inbetween.

There might be a way for us to tackle the handling more directly, but I haven't taken an eye to the code with that goal in mind yet.